src/main/java/com/icthh/xm/uaa/service/configurer/TenantManagerConfigurer.java

```diff
@@ -0,0 +1,17 @@
+package com.icthh.xm.uaa.service.configurer;
+
+import com.icthh.xm.commons.tenantendpoint.TenantManager;
+import com.icthh.xm.commons.tenantendpoint.provisioner.TenantProvisioner;
+
+public abstract class TenantManagerConfigurer {
```

Comment:
extract interface and use in all place interface. This implementation can be present, but by default must be interface.

Reply:
order of TenantProvisioner VERY important